src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,96 @@
+package duke.task;
+
+import java.util.regex.Pattern;
+
+import org.json.simple.JSONObject;
+
+import duke.date.Date;
+import duke.enums.Tasks;
+import duke.exception.InvalidDateException;
+import duke.exception.UnknownTaskTypeException;
+
+/** Represents a Task object. */
+public abstract class Task {
+    /** The description of the task */
+    protected String description;
+
+    /** The status of the task */
+    protected boolean isDone;
+
+    /** Default Task constructor */
+    protected Task() {
+        this("");
+    }
+
+    /**
+     * Task constructor.
+     *
+     * @param description The description of the Task.
+     */
+    protected Task(String description) {
+        this(description, false);
+    }
+
+    /**
+     * Task constructor.
+     *
+     * @param description the description of the task
+     * @param isDone the status of the task
+     */
+    protected Task(String description, boolean isDone) {
+        this.description = description;
+        this.isDone = isDone;
+    }
+
+    /** Marks the task's status as done */
+    public Task markTaskAsDone() {
```

Comment:
In my opinion, `markTaskAsDone()` sounds more natural than `setDone()`, as it is more akin to real-life task lists that you will "mark" tasks as done. But I understand `setDone()` sounds more OO because getters and setters or whatnot. Will consider changing this for the sake of other devs looking at my code. 👍🏼